### DISCUSSION TITLE
[FR] Esp32 Cam: add people detection / counting (= replacing & enhancing PIR)

### DISCUSSION THREAD

**Discussion by M0ebiu5** [arendst/Tasmota#10236] [Ideas]
It would be great, if the tasmota version for Esp32-Cam could detect people and maybe also count them.

Use case1: camera is mounted in the kitchen on the ceiling - if a person enters the area it sends a message also including the apr. coordinates. These will be used to determine which light or function has to be run.

Use case2: camera is mounted on the ceiling between rooms and maintains a counter for the people walking in and out. That should one enable to determine the number of people in a room more precisely than with a pir and also in near realtime.

Example (on esp32, the camera stream should be off/optional to lower cpu load):
[People counter](https://www.pyimagesearch.com/2018/08/13/opencv-people-counter/)

**Comment by ascillato2** [comment]
For people detection you need the power computing of a PC. For example, a raspberry is not enough for the AI recognition software. So, a tiny ESP32 is veeeery tiny to run any AI people recognition software. Sorry. Your request is impossible for being implemented in an ESP32.

You can use the ESP32CAM for just streaming the video to a PC and do the people recognition there and inform that to your home automation software.

**Reply by M0ebiu5** [reply]
Never say never :) Here is an example for
[arduino nano](https://www.hackster.io/little_lookout/person-detection-with-tensorflow-and-arduino-47ae01#overview)

**Comment by M0ebiu5** [comment]
Only detection will use a lot less resources than recognition - and there is already a face detection built in now...
I think it would even work (in most cases), if the esp32 only detects the movement of a group of pixel (without classifying). And the big advantage of the coordinates would persist.

**Comment by ascillato** [comment]
Cool. I wasn't aware that now TensorFlow has a Lite version ( https://github.com/tensorflow/tensorflow/tree/master/tensorflow/lite/micro/examples/person_detection/esp )

Thanks for sharing the links.

Few months ago, for using tensorflow in a Raspberry you needed to add an intel AI usb stick in order to improve the computational power of the RPi to make it work. Now they have simplified a lot the process but the training of the model still requires a huge amount of computing power ( they recommend google cloud for that due to every model requires several days )

**Reply by M0ebiu5** [reply]
Great, i didnt know that it already also exists for esp32!
Should be a piece of cake now :)))

**Reply by kugelkopf123** [reply]
As far as i know, is there already an integration of it. Search for Face on: https://tasmota.github.io/docs/Scripting-Language/
